core/execution/market.go

```diff
@@ -783,7 +783,7 @@ func (m *Market) blockEnd(ctx context.Context) {
 			m.releaseExcessMargin(ctx, closedWithoutLP...)
 		}
 		// last traded price should not reflect the closeout trades
-		m.lastTradedPrice = mp.Clone()
+		m.lastTradedPrice = m.getCurrentMarkPrice()
```

Comment:
to my eye, using `mp.Clone()` here looks safer. It'd be quite easy for changes in between `m.markPRice = mp` and this line to alter the mark price (eg closeout trades), so using the local variable ensures that doesn't happen

Reply:
Ok, changed now.